static/js/todo_widget.js

```diff
@@ -183,6 +228,22 @@ export function activate(opts) {
         const widget_data = task_data.get_widget_data();
         const html = render_widgets_todo_widget_tasks(widget_data);
         $elem.find("ul.todo-widget").html(html);
+        Sortable.create($elem.find("ul.todo-widget")[0], {
+            onUpdate(e) {
+                const old_pos = e.oldDraggableIndex;
+                const new_pos = e.newDraggableIndex;
+                // The way `idx` is assigned initially is such that it starts
+                // at 2 for a task list and increments by 2 for each task.
+                // For example, for a list with 3 tasks, the `idx` values would
+                // be 2, 4, and 6, while the actual indices in the tasks array
+                // would, of course, be 0, 1 and 2. Hence the addition by 1
+                // and doubling to convert from array index (`pos`) to `idx`
```

Comment:
This adds unnecessary complexity to the code. Can we keep the idx values and indices same?

Reply:
Since the idx values being the way they are, seem to have a lot of history and complexity, I've created [this thread](https://chat.zulip.org/#narrow/stream/49-development-help/topic/simplify.20todos.20indexing) to propose potentially changing them to be identical to indices (starting from 0 and incrementing by 1). Would really appreciate your opinions there!